ext/QuantumToolboxCairoMakieExt.jl

```diff
@@ -0,0 +1,214 @@
+module QuantumToolboxCairoMakieExt
+
+using QuantumToolbox
+using CairoMakie
+
+@doc raw"""
+    plot_wigner(
+        library::Val{:CairoMakie},
+        state::QuantumObject{DT,OpType};
+        xvec::Union{Nothing,AbstractVector} = nothing,
+        yvec::Union{Nothing,AbstractVector} = nothing,
+        g::Real = √2,
+        method::WignerSolver = WignerClenshaw(),
+        projection::Union{Val,Symbol} = Val(:two_dim),
+        location::Union{GridPosition,Nothing} = nothing,
+        colorbar::Bool = false,
+        kwargs...
+    ) where {DT,OpType}
+
+Plot the [Wigner quasipropability distribution](https://en.wikipedia.org/wiki/Wigner_quasiprobability_distribution) of `state` using the [CairoMakie](https://github.com/MakieOrg/Makie.jl/tree/master/CairoMakie) plotting library.
+
+Note that CairoMakie must first be imported before using this function.
+
+# Arguments
+- `library::Val{:CairoMakie}`: The plotting library to use.
+- `state::QuantumObject`: The quantum state for which the Wigner function is calculated. It can be either a [`KetQuantumObject`](@ref), [`BraQuantumObject`](@ref), or [`OperatorQuantumObject`](@ref).
+- `xvec::AbstractVector`: The x-coordinates of the phase space grid. Defaults to a linear range from -7.5 to 7.5 with 200 points.
+- `yvec::AbstractVector`: The y-coordinates of the phase space grid. Defaults to a linear range from -7.5 to 7.5 with 200 points.
+- `g::Real`: The scaling factor related to the value of ``\hbar`` in the commutation relation ``[x, y] = i \hbar`` via ``\hbar=2/g^2``.
+- `method::WignerSolver`: The method used to calculate the Wigner function. It can be either `WignerLaguerre()` or `WignerClenshaw()`, with `WignerClenshaw()` as default. The `WignerLaguerre` method has the optional `parallel` and `tol` parameters, with default values `true` and `1e-14`, respectively.
+- `projection::Union{Val,Symbol}`: Whether to plot the Wigner function in 2D or 3D. It can be either `Val(:two_dim)` or `Val(:three_dim)`, with `Val(:two_dim)` as default.
```

Comment:
how about `Val(:(2d))` and `Val(:(3d))` to make it more similar to `qutip`

Reply:
We had discarded this possibility at first because `:(2d)` and `:(3d)` are `Expr` objects, not `Symbol`s. The only way I could find to have 2d and 3d is via a string (e.g., `Symbol("2d")`). 
Are there other ways?

Reply:
I don't have other ways in mind. The current implementation is already fine for me